astropy/time/formats.py

```diff
@@ -370,15 +366,22 @@ def set_jds(self, val1, val2):
         self.jd1 = tm._time.jd1
         self.jd2 = tm._time.jd2
 
-    @property
-    def value(self):
-        # when we get here, getattr will already have ensured our scale
-        # equals epoch scale, so we can just subtract the epoch and convert
-        time_from_epoch = ((self.jd1 - self.epoch.jd1) +
-                           (self.jd2 - self.epoch.jd2)) / self.unit
+    def value(self, parent=None):
+        # Make sure that scale is the same as epoch scale so we can just
+        # subtract the epoch and convert
+        if self.scale != self.epoch_scale:
+            tm = parent.replicate()
```

Comment:
> Is there a reason not just to use `? (Or are you thinking ahead to when there will be a cache?).

Just the converse, I'm thinking of the case right now where `getattr(parent, self.epoch_scale)` always makes a replica even if the scale is not changing.  In the future when that is efficient then this code could be cleaned up.  Since I really want this particular bug fixed for my own work (I use `cxcsec`), I was doing the minimal change.